lib/libmpa/mpa_random.c

```diff
@@ -35,12 +35,12 @@
 #include "mpa_assert.h"
 #include <tee_api_types.h>
 
-/* This function ends to the real RNG of the platform
- * Depending where the libmpa is used:
- * - kernel side: the implementation is found in the TEECore
- * - user side: the implementation is found in libutee
- */
-TEE_Result get_rng_array(void *buf, size_t blen);
+static random_generator_cb get_rng_array;
+
+void mpa_set_random_generator(random_generator_cb callback)
+{
+	get_rng_array = callback;
+}
```

Comment:
Probably, but big ints can also be created by `TEE_BigIntConvertFromOctetString()` and `TEE_BigIntConvertFromS32()`.
In fact I think it's not good to have the RNG implemented in user space when we also have one in kernel. Especially if you consider that RNG initialization is not just setting a pointer: if you want proper entropy you have to load from saved state, then make sure the entropy pool is properly updated etc. This should better be handled entirely in the TEE core. So I would suggest that libmpa (user mode) should make a syscall when it needs random data.